server/src/main/java/io/druid/server/initialization/BaseJettyServerInitializer.java

```diff
@@ -51,4 +59,27 @@ private static void setDefaultGzipFilterHolderParameters(final FilterHolder filt
     // We don't actually have any precomputed .gz resources, and checking for them inside jars is expensive.
     filterHolder.setInitParameter("checkGzExists", String.valueOf(false));
   }
+
+  public void addExtensionFilters(ServletContextHandler handler, Injector injector) {
```

Comment:
This is more a comment on this class in general, but it appears to be an "abstract" class with the sole intention of supplying common functionality to its children.  Inheritance for this reason is often not the right choice of abstraction.

Generally, inheritance should be used when you have some stateful operations that you want to encapsulate and share amongst children implementations.  The parent class works on that common state and nothing else.

"Base class" inheritance can also be a useful tool in providing backwards compatible APIs giving you a place to do a default implementation of a method when new methods are added to an interface.

I don't believe either of those are happening in this case.  I think these are really just the equivalent of static methods and should be made as static methods.  If there is a need to override them for testing or something, then how about we create a class that does these things and then inject that instead?